Parse/src/main/java/com/parse/ParseInstallation.java

```diff
@@ -180,25 +184,65 @@ private void updateVersionInfo() {
         String appVersion = pkgInfo.versionName;
         String appName = pm.getApplicationLabel(pm.getApplicationInfo(packageName, 0)).toString();
 
-        if (packageName != null && !packageName.equals(get("appIdentifier"))) {
+        if (packageName != null && !packageName.equals(get(KEY_APP_IDENTIFIER))) {
           performPut(KEY_APP_IDENTIFIER, packageName);
         }
-        if (appName != null && !appName.equals(get("appName"))) {
+        if (appName != null && !appName.equals(get(KEY_APP_NAME))) {
           performPut(KEY_APP_NAME, appName);
         }
-        if (appVersion != null && !appVersion.equals(get("appVersion"))) {
+        if (appVersion != null && !appVersion.equals(get(KEY_APP_VERSION))) {
           performPut(KEY_APP_VERSION, appVersion);
         }
       } catch (PackageManager.NameNotFoundException e) {
         PLog.w(TAG, "Cannot load package info; will not be saved to installation");
       }
 
-      if (!VERSION_NAME.equals(get("parseVersion"))) {
+      if (!VERSION_NAME.equals(get(KEY_PARSE_VERSION))) {
         performPut(KEY_PARSE_VERSION, VERSION_NAME);
       }
     }
   }
 
+  /*
+   * Save locale in the following format:
+   *   [language code]-[country code]
+   *
+   * The language codes are two-letter lowercase ISO language codes (such as "en") as defined by
+   * <a href="http://en.wikipedia.org/wiki/ISO_639-1">ISO 639-1</a>.
+   * The country codes are two-letter uppercase ISO country codes (such as "US") as defined by
+   * <a href="http://en.wikipedia.org/wiki/ISO_3166-1_alpha-3">ISO 3166-1</a>.
+   *
+   * Note that Java uses several deprecated two-letter codes. The Hebrew ("he") language
+   * code is rewritten as "iw", Indonesian ("id") as "in", and Yiddish ("yi") as "ji". This
+   * rewriting happens even if you construct your own {@code Locale} object, not just for
+   * instances returned by the various lookup methods.
+   */
+  private void updateLocaleIdentifier() {
+    final Locale locale = Locale.getDefault();
+
+    String language = locale.getLanguage();
+    String country = locale.getCountry();
+
+    if (TextUtils.isEmpty(language)) {
+      return;
+    }
+
+    // rewrite depreciated two-letter codes
+    if (language.equals("iw")) language = "he"; // Hebrew
+    if (language.equals("in")) language = "id"; // Indonesian
+    if (language.equals("ji")) language = "yi"; // Yiddish
```

Comment:
Let's add a unit test to verify that we're correctly persisting these 3 locales.